rust/src/linalg/dot.rs

```diff
@@ -53,6 +55,13 @@ impl Dot for [f32] {
     type Output = f32;
 
     fn dot(&self, other: &[f32]) -> f32 {
+        #[cfg(target_arch = "x86_64")]
+        {
+            if is_x86_feature_detected!("fma") {
```

Comment:
do we need AVX2 flag here too?

Reply:
We dont not need. `fma` is one set newer to `avx2` , for the machine with  `fma`, all of them have `avx2`.